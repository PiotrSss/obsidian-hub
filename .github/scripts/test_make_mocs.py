# TODO Figure out how to move this to a tests directory, abd still import make_moc
from approvaltests import Options
from approvaltests.approvals import verify
from approvaltests.reporters import GenericDiffReporterFactory

from make_mocs import make_moc_for_directory, make_moc_for_sub_directories


def make_default_reporter():
    return Options().with_reporter(GenericDiffReporterFactory().get("AraxisMergeMac"))


def test_output_for_files():
    directory = '01 - Topic 1/01.02 Subtopic'
    files = ['README.md', 'publish.css', 'logo.svg', 'File 1.md', 'File 2.md']
    result = make_moc_for_directory(directory, files)
    verify(result)


def test_output_for_sub_directories():
    directory = '01 - Community'
    sub_directories = ['Authors - Persons', 'Events', 'Obsidian Roundup', 'Video Channels']
    result = make_moc_for_sub_directories(directory, sub_directories)
    verify(result, options=(make_default_reporter()))
