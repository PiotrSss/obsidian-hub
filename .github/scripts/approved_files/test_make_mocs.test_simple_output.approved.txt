- [[01 - Topic 1/01.02 Subtopic/File 1|File 1]]
- [[01 - Topic 1/01.02 Subtopic/File 2|File 2]]
